File: src/game/entity/enemies/FastEnemy.java
package game.entity.enemies;

import game.entity.models.Enemy;
import game.models.Coordinates;

import java.util.Observable;

// TODO
public class FastEnemy extends Enemy {
    public FastEnemy(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    public String[] getFrontIcons() {
        return new String[]{"assets/entities/enemies/yellow_ball/yellow_ball_0.png"};
    }

    @Override
    public String[] getLeftIcons() {
        return getFrontIcons();

    }

    @Override
    public String[] getBackIcons() {
        return getFrontIcons();
    }

    @Override
    public String[] getRightIcons() {
        return getFrontIcons();
    }

    @Override
    public float getSpeed() {
        return 1.5f;
    }
}

File: src/game/engine/world/domain/entity/actors/impl/enemies/npcs/FastEnemy.kt
package game.engine.world.domain.entity.actors.impl.enemies.npcs

import game.engine.world.types.EntityTypes
import game.engine.world.domain.entity.geo.Coordinates
import game.engine.world.domain.entity.geo.Direction
import game.engine.ui.panels.game.PitchPanel
import game.utils.Paths.enemiesFolder

/**
 * An enemy with increased speed multiplier;
 */
class FastEnemy : IntelligentEnemy {
    constructor() : super()
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    init {
        hitboxSizeToHeightRatio = 0.527f
    }

    override val entitiesAssetsPath: String get() = "$enemiesFolder/fast_enemy/fast_enemy"

    override fun getCharacterOrientedImages(): Array<String> = Array(4) { index ->
        "${entitiesAssetsPath}_${imageDirection.toString().lowercase()}_$index.png"
    }

    override fun getSpeed(): Float = 1f

    override fun getImageDirections(): List<Direction> = listOf(Direction.RIGHT, Direction.LEFT)

    override val size: Int
        get() = PitchPanel.COMMON_DIVISOR * 2

    override val type: EntityTypes
        get() = EntityTypes.FastEnemy
}